crates/binstalk-git-repo-api/src/gh_api_client/release_artifacts.rs

```diff
@@ -0,0 +1,193 @@
+use std::{
+    borrow::Borrow,
+    collections::HashSet,
+    fmt,
+    future::Future,
+    hash::{Hash, Hasher},
+};
+
+use binstalk_downloader::remote::{self};
+use compact_str::{CompactString, ToCompactString};
+use serde::Deserialize;
+use url::Url;
+
+use super::{
+    common::{issue_graphql_query, issue_restful_api},
+    GhApiError, GhRelease, GhRepo,
+};
+
+// Only include fields we do care about
+
+#[derive(Eq, Deserialize, Debug)]
+struct Artifact {
+    name: CompactString,
+    url: Url,
+}
+
+// Manually implement PartialEq and Hash to ensure it will always produce the
+// same hash as a str with the same content, and that the comparison will be
+// the same to coparing a string.
+
+impl PartialEq for Artifact {
+    fn eq(&self, other: &Self) -> bool {
+        self.name.eq(&other.name)
+    }
+}
+
+impl Hash for Artifact {
+    fn hash<H>(&self, state: &mut H)
+    where
+        H: Hasher,
+    {
+        let s: &str = self.name.as_str();
+        s.hash(state)
+    }
+}
+
+// Implement Borrow so that we can use call
+// `HashSet::contains::<str>`
+
+impl Borrow<str> for Artifact {
+    fn borrow(&self) -> &str {
+        &self.name
+    }
+}
+
+#[derive(Debug, Default, Deserialize)]
+pub(super) struct Artifacts {
+    assets: HashSet<Artifact>,
+}
+
+impl Artifacts {
+    /// get url for downloading the artifact using GitHub API (for private repository).
+    pub(super) fn get_artifact_url(&self, artifact_name: &str) -> Option<Url> {
+        self.assets
+            .get(artifact_name)
+            .map(|artifact| artifact.url.clone())
+    }
+}
+
+pub(super) fn fetch_release_artifacts_restful_api(
+    client: &remote::Client,
+    GhRelease {
+        repo: GhRepo { owner, repo },
+        tag,
+    }: &GhRelease,
+) -> impl Future<Output = Result<Artifacts, GhApiError>> + Send + Sync + 'static {
+    issue_restful_api(
+        client,
+        &[
+            "repos", owner, repo, "releases", "tags",
+            tag, //&percent_encode_http_url_path(tag).to_compact_string(),
```

Comment:
Rm this commented out code:

```suggestion
            tag,
```